### PR TITLE
Fix streamsb

### PR DESCRIPTION
**Pull request by shellc0de27**
This should do the trick

### CODE REVIEW

**Comment by shellc0de27** [comment]
So i noticed domain var (on line 47) was left from my testing. Which does work across all the ones i tested. However, should i update that to another one? What are your thoughts?

File: lib/resolveurl/plugins/streamsb.py [hunk 1]
```diff
@@ -43,9 +44,21 @@ def get_media_url(self, host, media_id):
             code, mode, hash = eval(helpers.pick_source(sources))
             dl_url = 'https://{0}/dl?op=download_orig&id={1}&mode={2}&hash={3}'.format(host, code, mode, hash)
             html = self.net.http_GET(dl_url, headers=headers).content
-            r = re.search('href="([^"]+)">Direct', html)
-            if r:
-                return r.group(1) + helpers.append_headers(headers)
+            domain = 'aHR0cHM6Ly9jbG91ZGVtYi5jb206NDQz'
+            token = helpers.girc(html, rurl, domain)
+            hash2 = re.search(r'hash"\s*value="([^"]+)', html)
```
Comment (Gujal00) [inline lib/resolveurl/plugins/streamsb.py]: line 49 not required as get_hidden function will collect the form data

File: lib/resolveurl/plugins/streamsb.py [hunk 2]
```diff
@@ -43,9 +44,21 @@ def get_media_url(self, host, media_id):
...
+            if token and hash2:
```
Comment (Gujal00) [inline lib/resolveurl/plugins/streamsb.py]: line 50 should just be
```python
if token:
```

File: lib/resolveurl/plugins/streamsb.py [hunk 3]
```diff
@@ -43,9 +44,21 @@ def get_media_url(self, host, media_id):
...
+                payload = {
+                    'op': 'download_orig',
+                    'id': code,
+                    'mode': mode,
+                    'hash': hash2.group(1),
+                    'g-recaptcha-response': token
+                }
```
Comment (Gujal00) [inline lib/resolveurl/plugins/streamsb.py]: Lines 51 to 57 can be replaced with
```python
                payload = helpers.get_hidden(html)
                payload.update({'g-recaptcha-response': token})
```

**Review by Gujal00 — CHANGES_REQUESTED** [review]
Thanks for the code, It can be slightly improved using the function to pull form data.
Please update the PR with the changes suggested  and test

**Comment by Gujal00** [comment]
> So i noticed domain var (on line 47) was left from my testing. Which does work across all the ones i tested. However, should i update that to another one? What are your thoughts?

Domain for this should be `aHR0cHM6Ly9zdHJlYW1zYi5uZXQ6NDQz`
it is basically base64 encode site base url, which is `https://streamsb.net:443` in this case

so the best way would be to replace that line with
```python
domain = base64.b64encode((rurl[:-1] + ':443').encode('utf-8'))
```

File: lib/resolveurl/plugins/streamsb.py (hunk 1 above)
Comment (shellc0de27) [reply lib/resolveurl/plugins/streamsb.py]: sweet thank you

File: lib/resolveurl/plugins/streamsb.py (hunk 2 above)
Comment (shellc0de27) [reply lib/resolveurl/plugins/streamsb.py]: got it :)

File: lib/resolveurl/plugins/streamsb.py (hunk 3 above)
Comment (shellc0de27) [reply lib/resolveurl/plugins/streamsb.py]: thanks

**Comment by shellc0de27** [comment]
For some reason i had better results just hard coding that domain you suggested

**Review by Gujal00 — APPROVED** [review]

**Comment by Gujal00** [comment]
Thanks

**Pull request merged**